mlir/lib/Dialect/Rock/Tuning/RockTuningImpl.cpp

```diff
@@ -151,20 +188,29 @@ TunableParams *createTunableParamSpace(ModuleOp &mod) {
   return newSpace;
 }
 
-bool tuningGetParam(TunableParams *tuningSpace, int pos,
-                    ParamEntry *paramEntry) {
+bool tuningGetParamFull(TunableParams *tuningSpace, unsigned pos,
+                        ParamEntry *paramEntry) {
+  // out of bound check.
+  if (pos > tuningSpace->tuningRangeFull.size() - 1)
+    return false;
+  paramEntry->param = tuningSpace->tuningRangeFull[pos];
+  return true;
+}
+
+bool tuningGetParamQuick(TunableParams *tuningSpace, unsigned pos,
+                         ParamEntry *paramEntry) {
   // out of bound check.
-  if (pos < 0 || (unsigned int)pos > tuningSpace->tuningRange.size() - 1)
+  if (pos > tuningSpace->tuningRangeFull.size() - 1)
```

Comment:
This seems not right...

Reply:
The "buffer to be filled" thing was @pfultz2's request - I was going to go for `char **` arguments so we `malloc()` and return the buffer, but he was wanting to avoid the extra copy

Reply:
... right, comment on the wrong thing.

This was a typo, fixed